### PR TITLE
##161342856 bookmark articles

### PR DESCRIPTION
**Pull request by wiztemple**
#### What does this PR do?
- Bookmarks articles

#### Description of tasks to be completed
- create bookmark model
- create Bookmark controller
- create a method to get a single article
- create a method to bookmark articles
- test bookmark endpoint
- document endpoint

#### What are the relevant Pivotal Tracker stories?
- [#161342856](https://www.pivotaltracker.com/story/show/161342856)

#### How can this be manually tested?
- git clone https://github.com/andela/haven-ah-backend.git.
- cd into the branch ft-bookmark-articles-161342856
- run npm install to install the npm packages.
- run npm test to run the unit and integration tests.

#### Screenshots(if appropriate)
screenshot 2018-11-05 at 5 43 27 pm
screenshot 2018-11-05 at 5 43 53 pm
screenshot 2018-11-06 at 5 19 58 pm

### CODE REVIEW

File: server/config/config.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ const config = {
     port: process.env.DB_PORT,
     PORT: 5000,
     dialect: 'postgres',
-    secret: process.env.JWT_SECRET
+    secret: process.env.JWT_SECRET,
+    logging: false
```
Comment (seyi-adeleke) [inline server/config/config.js]: why was this change required? Shouldn't we be logging in dev?

File: server/controllers/bookmark.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+class Bookmark {
+  /**
+   * bookmark an article
+   * @param {object} request Request Object
+   * @param {object} response Response Object
+   * @returns {object} User Object
+   */
+  static async bookmarkArticle(request, response) {
+    const { slug } = request.params;
```
Comment (seyi-adeleke) [inline server/controllers/bookmark.js:15]: what happens if you pass an invalid data type as the slug?

File: server/controllers/bookmark.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    const { userId } = request;
+    const findArticle = await articleRepo.getSingleArticle(slug);
```
Comment (seyi-adeleke) [inline server/controllers/bookmark.js]: and if slug is null here?

File: server/repository/articleRepository.js [hunk 4]
```diff
@@ -55,6 +55,24 @@ class ArticleRepository {
...
     return articleRecords;
   }
+
+  /**
+   * Function to get all article in the database
+   * @param { string } slug
+   * @returns { object }
+   ** otherwise it throws an error
+   */
+  static async getSingleArticle(slug) {
```
Comment (seyi-adeleke) [inline server/repository/articleRepository.js]: can this be getArticleBySlug?

File: server/repository/articleRepository.js [hunk 5]
```diff
@@ -55,6 +55,24 @@ class ArticleRepository {
     articleRecords.meta = getPaginationMeta(limit, page, articleRecords.count);
     return articleRecords;
   }
+
+  /**
+   * Function to get all article in the database
+   * @param { string } slug
+   * @returns { object }
+   ** otherwise it throws an error
```
Comment (seyi-adeleke) [inline server/repository/articleRepository.js:63]: This jsdoc isnt correct

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]
Good job. please add additional tests as you've dropped our current coverage

File: server/repository/articleRepository.js (hunk 4 above)
Comment (wiztemple) [reply server/repository/articleRepository.js]: okay

File: server/config/config.js (hunk 1 above)
Comment (wiztemple) [reply server/config/config.js]: I forgot to remove it, I wanted to use it to test something.

File: server/controllers/bookmark.js (hunk 2 above)
Comment (PriscillaSam) [reply server/controllers/bookmark.js:15]: @seyi-adeleke  we have a param validator that @wiztemple can use.

Comment (wiztemple) [reply server/controllers/bookmark.js:15]: it returns article not found.

File: server/controllers/bookmark.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+   * @returns {object} User Object
+   */
+  static async bookmarkArticle(request, response) {
+    const { slug } = request.params;
+    const { userId } = request;
+    const findArticle = await articleRepo.getArticleBySlug(slug);
+    if (!findArticle) {
+      return badHttpResponse(response, 404, 'article not found');
+    }
+    const bookmarked = await bookmarkRepo.createBookmark(userId, findArticle.id);
```
Comment (seyi-adeleke) [inline server/controllers/bookmark.js:21]: do we need to send the whole bookmark object?
Also this should be `bookmark` not `bookmarked`

**Review by seyi-adeleke — APPROVED** [review]
